File: zepz/Stride/temp.py
class Node:
    def __init__(self, data: int = None, next: 'Node' = None) -> None:
        self.data = data
        self.next = next

    def add(self, data) -> 'Node':
        new_node = Node(data)

        current_node = self

        while (current_node.next != None):
            current_node = current_node.next

        current_node.next = new_node
        return self

    def __repr__(self) -> str:
        current_node = self
        result = ''

        while current_node is not None:
            result += f'{current_node.data}'

            if current_node.next is not None:
                result += ' --> '

            current_node = current_node.next

        return result


l = Node(1)
l.add(2).add(3)

print(l)

File: zepz/Stride/temp.js
class Node {
  constructor(data) {
    this.data = data;
    this.next = null;
  }
}

class LinkedList {
  constructor() {
    this.head = null;
  }

  add(data) {
    let newNode = new Node(data);

    if (this.head == null) {
      this.head = newNode;
      return this;
    }

    let currNode = this.head;

    while (currNode.next != null) {
      currNode = currNode.next;
    }

    currNode.next = newNode;
    return this;
  }

  pop(data) {
    let currNode = this.head;

    if (currNode == null) {
      throw Error("List if Emprty");
    }

    while (currNode != null) {
      if (currNode.data == data) {
        currNode = currNode.next;
      }
      curr;
    }
  }

  toString() {
    let currNode = this.head;
    let result = "";

    while (currNode) {
      result += `${currNode.data}`;

      if (currNode.next != null) {
        result += " --> ";
      }

      currNode = currNode.next;
    }

    return result;
  }
}

let ll = new LinkedList();
console.log(ll.toString());

ll.add(1).add(2).add(3);
console.log(ll.toString());
